Update readme to reflect repo

rudyflores opened the pull request:
**What It Does**
<!-- A list of relevant issues, enhancements, fixed bugs, etc -->
Update readme to properly reflect how to use and install the project.

**How to Test**
<!-- If a bug has been fixed, how can reviewers verify that the change(s) fixed it? -->

**Review Checklist**
I certify that I have:
- [ ] tested my changes
- [ ] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)


**Additional Comments**
<!-- Anything else noteworthy about this pull request. This section is optional. -->

t1m0thyj commented on .editorconfig:
```diff
@@ -3,7 +3,7 @@ root = true
 [*]
 indent_style = space
-indent_size = 4
+indent_size = 2
```
Is it intentional to change the indent from 2 to 4? All our source files are still using 4 spaces, and I think this is consistent with other projects like ZE.

rudyflores replied:
I saw some warnings on the readme when performing 4 indent vs 2 indent for some reason, I'm not sure if there is some issue with how the linter is viewing this vs how many tabs are made which led me to change this...

rudyflores replied:
I switched it back to 4, perhaps we can see later why that warning for indents shows with 4 on md files

t1m0thyj commented on README.md:
```diff
@@ -1,239 +1,99 @@
+# Kubernetes Secrets for Zowe CLI & Zowe Explorer
 
-<img src="imgs/Zowe_ConformanceBadge_cli.png" width=25% alt="Zowe Conformance Badge"/>
+The Kubernetes Secrets plugin for Zowe CLI and Zowe Explorer allows you to store your Zowe profile credetials as Kubernetes secrets in your Kubernetes cluster of choice. This is by changing the default credential manager upon installation into Zowe CLI or Zowe Explorer.
```
`credetials` -> `credentials`

anaxceron replied:
please hyphenate all uses of the word **plug-in** so its consistent w/ Zowe Docs

t1m0thyj commented on README.md:
````diff
@@ -1,239 +1,99 @@
 
-**Follow these steps:**
+   ```bash
+   zowe plugins uninstall @zowe/secure-credential-store-for-zowe-cli
````
Should the plugin name be updated to the K8s plugin?

anaxceron commented on README.md:
```diff
@@ -1,239 +1,99 @@
+## Requirements
 
-## How the plug-in works
+Before you install and use the plug-in make sure you have the following installed:
```
please define what needs to be installed -- is it "applications"? is it "items"? Rewrite to say: Make sure you have the following applications/items/whatever installed:"

anaxceron commented on README.md:
```diff
@@ -1,239 +1,99 @@
-  - `gcc-c++`
-  - `libsecret-devel`
+- Zowe CLI on your computer. For more information, see [installing Zowe CLI](https://docs.zowe.org/stable/user-guide/cli-installcli.html).
+- Install [Node.js](https://nodejs.org/en/download) v16.0 or later.
```
- I don't think it's necessary to include "on your computer"
- Remove the verb "install" b/c this is a list of items (according to the sentence in Line 15), not actions.

anaxceron commented on README.md:
```diff
@@ -1,239 +1,99 @@
-To define access credentials to the server, copy the file named `.../__tests__/__resources__/properties/example_properties.yaml` and create a file named `.../__tests__/__resources__/properties/custom_properties.yaml`.
-
-**Note:** Information about how to customize the `custom_properties.yaml` file is provided in the yaml file itself.
+When an unsuccessful message displays, you can troubleshoot the installation by addressing the issues that the message describes. You can also review the information that is contained in the log file that is located in the Zowe CLI home directory.
```
- is it an "unsuccessful" message or is it a "failure" message?
- I would reword the sentence: " ... by addressing the issues outlined/described in the message."
- Also would tweak the next sentence: "... contained in the log file located in the Zowe CLI ..."

rudyflores replied:
it is a failure message that is displayed when something went wrong when installing the plug-in :)

anaxceron commented on packages/cli/CHANGELOG.md:
```diff
@@ -5,3 +5,4 @@ All notable changes to the Kubernetes Secrets Credential Manager Plug-in for Zow
 ## Recent Changes
 
 - Added system test for v2 profiles
+- Added readme with instructions on how to use and install the repo
```
Please add a link to the related issue or PR so that we can included that in the release notes

anaxceron reviewed (approved):
Thanks for working on this, Rudy! Left some comments, most of them regarding style.

rudyflores commented:
Thanks for the very helpful feedback @anaxceron @t1m0thyj !

anaxceron commented on README.md:
```diff
@@ -1,239 +1,99 @@
-  - `make`
-  - `gcc-c++`
-  - `libsecret-devel`
+- Zowe CLI For more information, see [installing Zowe CLI](https://docs.zowe.org/stable/user-guide/cli-installcli.html).
```
Needs a period between "Zowe CLI" and "For more information"

anaxceron reviewed (approved):
Looks great, Rudy -- just one minor typo that I pointed out in a comment. Thank you!

JillieBeanSim reviewed (commented):
Thanks @rudyflores for putting the readme together.  Since it is top level readme should it include installation/build instructions for both the cli and vsc ext?  Will you be adding a readme for the separate packages?  I think they are used in the marketplaces (vsc & npm) for package description and how to's

rudyflores commented:
@JillieBeanSim updates were made now, I added a top level readme with common instructions between both repos, and inside each repo (vscode and cli) I have added a readme that addresses installation and other cases for each one properly.

traeok commented on README.md:
```diff
@@ -1,239 +1,23 @@
--   [Running tests](#running-tests)
--   [Uninstalling](#uninstalling)
--   [Contributing](#contributing)
+The Kubernetes Secrets plug-in for Zowe CLI and Zowe Explorer allows you to store your Zowe profile credentials as Kubernetes secrets in your Kubernetes cluster of choice. This is by changing the default credential manager upon installation into Zowe CLI or Zowe Explorer.
```
```suggestion
The Kubernetes Secrets plug-in for Zowe CLI and Zowe Explorer allows you to store your Zowe profile credentials as Kubernetes secrets in your Kubernetes cluster of choice. It changes the default credential manager when installed in Zowe CLI or Zowe Explorer.
```

traeok commented on README.md:
```diff
@@ -1,239 +1,23 @@
--   [Contributing](#contributing)
+The Kubernetes Secrets plug-in for Zowe CLI and Zowe Explorer allows you to store your Zowe profile credentials as Kubernetes secrets in your Kubernetes cluster of choice. This is by changing the default credential manager upon installation into Zowe CLI or Zowe Explorer.
 
+This plug-in can also be used as a sample to build your own custom credential manager to use based on your own use case.
```
```suggestion
This plug-in also serves as a custom credential manager sample for your own use cases.
```

traeok commented on README.md:
````diff
@@ -1,239 +1,23 @@
-    export $(dbus-launch)
-    gnome-keyring-daemon -r --unlock --components=secrets
-    ```
+In addition, when storing Kubernetes secrets, the secret will be stored in the assigned namespace for the current context or otherwise through an environment variable named `$DEVWORKSPACE_NAMESPACE` and `$WORKSPACE_NAMESPACE`.
````
```suggestion
In addition, when storing Kubernetes secrets, users can specify a namespace through the environment variables `$DEVWORKSPACE_NAMESPACE` and `$WORKSPACE_NAMESPACE`. Otherwise, the secret will be stored in the assigned namespace for the current context.
```

traeok commented on README.md:
```diff
@@ -1,239 +1,23 @@
+## Requirements
 
-    **Note:** The following steps have been tested on CentOS, SUSE, and Ubuntu. Results may vary on other Linux distributions.
+Before you install and use the plug-in make sure you have the following tools installed:
```
missing comma between "plug-in" and "make sure"

traeok reviewed (approved):
LGTM, thanks @rudyflores - left a couple suggestions 👍

traeok reviewed (approved):
LGTM, thanks @rudyflores!

anaxceron commented on packages/cli/README.md:
```diff
@@ -0,0 +1,76 @@
+
+- Install the plug-in to Zowe CLI from an online registry or a local package.
+
+  Use this method when you want to install the plug-in quickly and start using it. With this method, you secure your user IDs and passwords manually.
```
When you say, "Use this method," which method are you referring to, the online registry or the local package? Or do you mean to say, "Use these methods when ..."
Should we split the online registry and local package into two methods? It might be a little confusing to have two different options w/in one method.

rudyflores replied:
In this case we point to external docs for more information on both methods, I meant to say "Use these methods when ..." as a disclamer before showing the link to external docs, so I think a separation is not necessary

anaxceron commented on packages/vscode/README.md:
```diff
@@ -0,0 +1,53 @@
+
+- Install the plug-in to VS Code from the marketplace
+
+  1. On the left most side of the VS Code window in your workspace, click on the extensions icon
```
- Step 1 and Step 3 should end with periods.
- Any time you refer to a specific item in the UI, I would capitalize it and bold it: Click on the **Extensions** icon; Click on the Search bar; Click on the Install button. It's just a good way to let the reader know that you're referring to a specific thing on the UI.

anaxceron commented on packages/vscode/README.md:
```diff
@@ -0,0 +1,53 @@
+
+To build from source, follow these steps:
+
+1. Clone this repository
```
Any steps that are a complete sentence, like this one, should end w/ a  period.

anaxceron commented on packages/vscode/README.md:
````diff
@@ -0,0 +1,53 @@
+     npm install && npm build && npm package
+   ```
+
+1. View the `dist/` folder in the project, this should contain a file named `Zowe.secrets-for-kubernetes.vsix`
````
I would tweak this sentence a little. Either:

- View the `dist/` folder in the project. This should contain ...
- View the `dist/` folder in the project, which should contain ...

anaxceron commented on packages/vscode/README.md:
```diff
@@ -0,0 +1,53 @@
+
+1. On top of the extensions view, click on the search bar and search for `Kubernetes Secrets for Zowe Explorer`
+
+1. click uninstall
```
All the Uninstalling steps should end w/ periods.
